junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/AbstractExtensionContext.java

```diff
@@ -89,9 +81,31 @@ abstract class AbstractExtensionContext<T extends TestDescriptor> implements Ext
 				.map(TestTag::getName)
 				.collect(collectingAndThen(toCollection(LinkedHashSet::new), Collections::unmodifiableSet));
 		// @formatter:on
+
+		this.closeResources = createCloseResources();
+		this.valuesStore = createStore(parent, launcherStoreFacade);
+	}
+
+	@SuppressWarnings("deprecation")
+	private NamespacedHierarchicalStore.CloseAction<org.junit.platform.engine.support.store.Namespace> createCloseResources() {
+		return (__, ___, value) -> {
+			if (value instanceof Store.CloseableResource) {
```

Comment:
We need to first check for `AutoCloseable` (if enabled) and log a warning if `CloseableResource` is implemented but not `AutoCloseable` (unless the feature is disabled).